src/search.c

```diff
@@ -263,12 +379,58 @@ static bool file_search(char *filename, char *path, char *search_term)
 	return false;
 }
 
-/* report_find()
+double guess_score(char *bundle, char *file, char *search_term)
+{
+	double multiplier = 1.0;
+	double score = 1.0;
+
+	char bnb[PATH_MAX];
+	char dnb[PATH_MAX];
+
+	char *bn;
+	char *dn;
+
+	strcpy(bnb, file);
+	strcpy(dnb, file);
+
+	bn = basename(bnb);
+	dn = dirname(dnb);
+
+
+	if (strstr(bundle, "-dev"))
+		multiplier /= 10;
+
+	if (strcmp(dn, "/usr/bin") == 0)
+		score += 5;
+
+	if (strcmp(dn, "/usr/include") == 0)
+		multiplier /= 2;
+
+	if (strstr(bundle, search_term))
+		score += 0.5;
+
+	/* slightly favor shorter filenames */
+	score += (2.0 / strlen(file));
```

Comment:
Division by zero if you say 

      swupd search ''

?

Reply:
That is an invalid search string actually.

Reply:
Oh, this isn't even the search term, this is the file name. We shouldn't have zero-length file names (`/` would be the shortest).